src/main/java/uk/gov/ons/fwmt/job_service/service/impl/FileIngestServiceImpl.java

```diff
@@ -25,115 +31,32 @@
-
-    // Timestamp validation
-    String rawTimestamp = filenameSplitByUnderscore[filenameSplitByUnderscore.length - 1];
-    LocalDateTime timestamp = getLocalDateTime(rawFilename, rawTimestamp);
-    log.debug("Timestamp detected as {}", timestamp);
-
-    Filename filename = new Filename(endpoint, tla, timestamp);
-    log.debug("Passed a filename check with {}", filename);
-    return filename;
-  }
-
-  protected String extractEndpoint(String rawFilename, String expectedEndpoint, String[] filenameSplitByUnderscore) {
-    String endpoint = filenameSplitByUnderscore[0];
-
-    if (!expectedEndpoint.equals(endpoint)) {
-      throw FWMTCommonException
-          .makeInvalidFileNameException(rawFilename, "File had an incorrect endpoint of " + endpoint);
-    }
-
-    switch (endpoint) {
-    case "staff":
-      if (filenameSplitByUnderscore.length != 2) {
-        throw FWMTCommonException
-            .makeInvalidFileNameException(rawFilename, "File names for staff should contain one underscore");
-      }
-      break;
-    case "sample":
-      if (filenameSplitByUnderscore.length != 3) {
-        throw FWMTCommonException
-            .makeInvalidFileNameException(rawFilename, "File names for samples should contain two underscores");
-      }
-      break;
-    default:
-      throw new IllegalArgumentException("File had an unrecognized endpoint of " + endpoint);
-    }
-    return endpoint;
+  private CSVParsingService csvParsingService;
+  
+  @Autowired
+  public FileIngestServiceImpl(CSVParsingService csvParsingService) {
+    this.csvParsingService = csvParsingService;
   }
-
-  protected String[] checkFileExtension(String rawFilename) {
-    String[] filenameSplitByDot = rawFilename.split("\\.");
-    if (filenameSplitByDot.length != 2 || !("csv".equals(filenameSplitByDot[1]))) {
-      throw FWMTCommonException.makeInvalidFileNameException(rawFilename, "No 'csv' extension");
-    }
-    return filenameSplitByDot;
-  }
-
-  protected LegacySampleSurveyType getLegacySampleSurveyType(String[] filenameSplitByUnderscore, String endpoint) {
-    LegacySampleSurveyType tla = null;
-    if (endpoint.equals("sample")) {
-      String tlaString = filenameSplitByUnderscore[1].toUpperCase();
-      log.debug("File TLA detected as " + tlaString);
-      switch (tlaString) {
-      case "LFS":
-        tla = LegacySampleSurveyType.LFS;
-        break;
-      case "GFF":
-        tla = LegacySampleSurveyType.GFF;
-        break;
-      default:
-        throw new IllegalArgumentException("File had an unrecognized TLA of " + tlaString);
+  
+  public SampleSummaryDTO validateSampleFile(File file) throws IOException{
+    SampleFilenameComponents filename = SampleFileUtils.buildSampleFilenameComponents(file);
+    final Reader reader = new InputStreamReader(new FileInputStream(file), StandardCharsets.UTF_8);
+    
+    Iterator<CSVParseResult<LegacySampleIngest>> csvRowIterator = csvParsingService.parseLegacySample(reader, filename.getTla());
```

Comment:
Above 3 lines are called twice, do we need this? possible to make this call once saving some time.

Reply:
As discussed Ill leave this as is for the moment until we relook at this area